sdk/servicebus/azure-servicebus/azure/servicebus/_common/_connection_string_parser.py

```diff
@@ -71,39 +67,14 @@ def parse_connection_string(conn_str):
     :type conn_str: str
     :rtype: ~azure.servicebus.ServiceBusConnectionStringProperties
     """
-    conn_settings = [s.split("=", 1) for s in conn_str.split(";")]
-    if any(len(tup) != 2 for tup in conn_settings):
-        raise ValueError("Connection string is either blank or malformed.")
-    conn_settings = dict(conn_settings)
-    shared_access_signature = None
-    for key, value in conn_settings.items():
-        if key.lower() == "sharedaccesssignature":
-            shared_access_signature = value
-    shared_access_key = conn_settings.get("SharedAccessKey")
-    shared_access_key_name = conn_settings.get("SharedAccessKeyName")
-    if any([shared_access_key, shared_access_key_name]) and not all(
-        [shared_access_key, shared_access_key_name]
-    ):
-        raise ValueError(
-            "Connection string must have both SharedAccessKeyName and SharedAccessKey."
-        )
-    if shared_access_signature is not None and shared_access_key is not None:
-        raise ValueError(
-            "Only one of the SharedAccessKey or SharedAccessSignature must be present."
-        )
-    endpoint = conn_settings.get("Endpoint")
-    if not endpoint:
-        raise ValueError("Connection string is either blank or malformed.")
-    parsed = urlparse(endpoint.rstrip("/"))
-    if not parsed.netloc:
-        raise ValueError("Invalid Endpoint on the Connection String.")
-    namespace = parsed.netloc.strip()
+    namespace, policy, key, entity, signature = _parse_conn_str(conn_str)[:-1]
+    endpoint = "sb://" + namespace + "/"
```

Comment:
could this be returned from `_parse_conn_str` instead of we manually rebuild it?

Reply:
Yes it can be returned. However, it would also require every other call to _parse_conn_str, including the all of the clients, to be changed as well. I'm not sure whether this would be worth the trade off, especially since this is the only method that would use the returned endpoint value and all the other ones would need to receive it and not store it.

Reply:
you're more considerate than me :P
I think we're good for now